data_models.py

```diff
@@ -211,6 +230,17 @@ def write_models(self, output_dir, file_suffix, test_set=False):
                 pca_df = self.pca_df
             pca_df.to_csv(output_file, sep='\t', compression='gzip')
 
+        if hasattr(self, 'ica_df'):
```

Comment:
I feel like you could combine these write_file sections into one to avoid repeating yourself. Just pass in a couple more parameters, like what to add to the filename. Your decision ultimately, but it seems like you're pretty close to making the write process generalizable across models anyway.

Reply:
I definitely agree, this code could be made a lot cleaner. I think it's a non-trivial refactor though, and I'd like to set up regression tests to make sure I'm not breaking things while I refactor, so I'm going to create a separate PR for this. Let me know if you disagree.

Created greenelab/netreg#8 to remind myself.

Reply:
I think that's a great idea.